doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,260 @@
+``*``; new parameters do not have to be added after the existing parameters.
+
+
+Test functions from `numpy.testing`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In new code, don't use `assert_almost_equal`, `assert_approx_equal` or
+`assert_array_almost_equal`. This is from the docstrings of these
+functions::
+
+    It is recommended to use one of `assert_allclose`,
+    `assert_array_almost_equal_nulp` or `assert_array_max_ulp`
+    instead of this function for more consistent floating point
+    comparisons.
+
+
+Testing that expected exceptions are raised
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When writing a new test that a function call raises a certain exception,
+the preferred style is to use ``pytest.raises`` as a context manager, with
+the code that is supposed to raise the exception in the code block defined
+by the context manager.  The `match` keyword argument of ``pytest.raises``
+is given with enough of the expected error message attached to the exception
+to ensure that the expected exception is raised.
+
+For example, the function `scipy.stats.zmap` is supposed to raise a
+``ValueError`` if the input contains ``nan`` and ``nan_policy`` is ``"raise"``.
+A test for this is::
+
+    scores = np.array([1, 2, 3])
+    compare = np.array([-8, -3, 2, 7, 12, np.nan])
+    with pytest.raises(ValueError, match='input contains nan'):
+        stats.zmap(scores, compare, nan_policy='raise')
+
+The ``match`` argument ensures that the test doesn't pass by raising
+a ``ValueError`` that is not related to the input containing ``nan``.
+
+
+A parameter called lambda
+~~~~~~~~~~~~~~~~~~~~~~~~~
+There are functions where convention dictates that (ideally) a parameter
+should be called 'lambda'.  In Python, 'lambda' is a reserved word that
+cannot be used as a variable name.  If a new function or class has a
+parameter where a name that is similar to 'lambda' is really the best
+choice, the preferred name to use is [TBD: ``lam``, ``lmb``, ``lamb``,
+``lmbda``, ``lambda_``?].
+
+There has been a lack of consistency in the past:
+
+============================== =============
+Function or class              Variable name
+============================== =============
+`scipy.stats.boxcox`           ``lmbda``
+`scipy.stats.boxcox_llf`       ``lmb``
+`scipy.stats.yeojohnson`       ``lmbda``
+`scipy.stats.yeojohnson_llf`   ``lmb``
+`scipy.stats.chi2_contingency` ``lambda_``
+`scipy.stats.boltzmann`        ``lambda_``
+`scipy.stats.planck`           ``lambda_``
+`scipy.stats.tukeylambda`      ``lam``
+`scipy.signal.cspline1d`       ``lamb``
+`scipy.signal.qspline1d`       ``lamb``
+`scipy.signal.cspline2d`       ``lambda`` [*]
+`scipy.signal.qspline2d`       ``lambda`` [*]
+`scipy.signal.spline_filter`   ``lmbda``
+`scipy.special.tklmbda`        ``lmbda``
+`scipy.special.boxcox`         ``lmbda``
+`scipy.special.boxcox1p`       ``lmbda``
+`scipy.special.inv_boxcox`     ``lmbda``
+`scipy.special.inv_boxcox1p`   ``lmbda``
```

Comment:
oh dear!

Reply:
Yeah, it looks bad when seen all at once like this, but a user reading the docstring of one of these functions would probably never notice the inconsistency.  I'm removing this section, but we can revisit the topic if there is interest.